From 9a4d3cadc4da431d9f103c41c2b78b6de242158f Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Thu, 23 Apr 2020 10:37:21 -0700
Subject: [PATCH 4/7] misc: mrvl-loki: Add function to unregister the irq
 callback

commit 87bdb5989427163035d536cebdd2a020ef32ccd6 from
git@git.assembla.com:cavium/WindRiver.linux.git

The function to unregister the irq callback helps in avoiding a kernel
panic if an interrupt arrives and the driver with cabllback is rmmod.

Change-Id: Ia46ac1a67e69bc081026115930f4bd51d1a4cba5
Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/27413
Reviewed-by: Stanislaw Kardach <Stanislaw.Kardach@cavium.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/misc/mrvl-loki.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/misc/mrvl-loki.c b/drivers/misc/mrvl-loki.c
index c2757f1fe4c2..258cf5f4a46f 100644
--- a/drivers/misc/mrvl-loki.c
+++ b/drivers/misc/mrvl-loki.c
@@ -58,6 +58,12 @@ int mrvl_loki_register_irq_cb(int (*func)(uint32_t instance, uint32_t pss_int))
 }
 EXPORT_SYMBOL(mrvl_loki_register_irq_cb);
 
+void mrvl_loki_unregister_irq_cb(void)
+{
+	g_ml->irq_cb = NULL;
+}
+EXPORT_SYMBOL(mrvl_loki_unregister_irq_cb);
+
 static irqreturn_t mrvl_loki_handler(int irq, void *dev)
 {
 	struct mrvl_loki *ml =
-- 
2.17.1

